scapy/contrib/roce.py

```diff
@@ -188,10 +209,34 @@ def compute_icrc(self, p):
     def post_build(self, p, pay):
         # type: (bytes, bytes) -> bytes
         p += pay
+        if self.opcode is None:
+            opcode = self.guess_opcode() or 0
+            p = struct.pack('B', opcode) + p[1:]
```

Comment:
Use `bind_layers(BTH, DETH, opcode=[number])` instead of a `post_build` and a `guess_payload`

Reply:
Is it possible to look two layers below with `bind_layers`? In this case, the UD_SEND_WITH_IMM opcode is used when having both the DETH header and the ImmDt afterward this BTH header.

Reply:
No it's not

Reply:
So what is the recommended way to handle such packets? Would it be better to have just one big header type for RoCE with conditional fields maybe? Or just give up on automatically picking the opcode field?

Reply:
Hi, this time I'm sorry for the long delay :) 
I've tried replacing the `guess_payload` use with conditional fields. Please let know what you think.